apps/api/src/app/workflows-v2/mappers/notification-template-mapper.ts

```diff
@@ -14,34 +12,31 @@ import {
   WorkflowTypeEnum,
 } from '@novu/shared';
 import { NotificationStepEntity, NotificationTemplateEntity } from '@novu/dal';
-import { GetPreferencesResponseDto } from '@novu/application-generic';
+import { GetWorkflowByIdsResponseDto } from '@novu/application-generic';
 import { buildSlug } from '../../shared/helpers/build-slug';
 
-export function toResponseWorkflowDto(
-  template: NotificationTemplateEntity,
-  preferences: GetPreferencesResponseDto | undefined
-): WorkflowResponseDto {
+export function toResponseWorkflowDto(workflow: GetWorkflowByIdsResponseDto): WorkflowResponseDto {
   const preferencesDto: PreferencesResponseDto = {
-    user: preferences?.source[PreferencesTypeEnum.USER_WORKFLOW] || null,
-    default: preferences?.source[PreferencesTypeEnum.WORKFLOW_RESOURCE] || DEFAULT_WORKFLOW_PREFERENCES,
+    user: workflow.userPreferences,
+    default: workflow.defaultPreferences,
```

Comment:
Mostly renaming, this is the only functional change to start reading from the internal Workflow DTO.